### DISCUSSION TITLE
Can an (XPath) rule check the current file name?

### DISCUSSION THREAD

**Discussion by rsoesemann** [pmd/pmd#3444] [Q&A] [answered]
I want to use XPath rules to check whether files name comply with certain naming rules? Is that possible in PMD?

**Comment by oowekyala** [comment]
Not possible with xpath rules yet. We could add an xpath function to fetch the file name though

**Reply by rsoesemann** [reply]
Cool. Is there an example PR that shows how much work this would be?

**Reply by oowekyala** [reply]
I don't think so. Actually I think it would be simpler to add a method `getFileName` to apex root nodes, which would be usable as an attribute. The field would be set by the parser. The file name wouldn't otherwise be accessible in the implementation of an xpath function iirc

**Reply by rsoesemann** [reply]
I think this wouldn't only be an Apex addition. I personally need it for XML files and Xpath rules.

**Comment by adangel** [comment, marked as answer]
With PMD 6.38.0 and later, there is a new XPath function "pmd:fileName()" which returns the current filename - see https://docs.pmd-code.org/snapshot/pmd_userdocs_extending_writing_xpath_rules.html#pmd-fileName for the docs.
